client/my-sites/themes/single-site-jetpack.jsx

```diff
@@ -38,8 +42,11 @@ const ConnectedThemesSelection = connectOptions(
 				getOptions={ function( theme ) {
 					return pickBy(
 						addTracking( props.options ),
-						option => ! ( option.hideForTheme && option.hideForTheme( theme, props.siteId ) )
-					); } }
+						( option, key ) => ! ( option.hideForTheme && option.hideForTheme( theme, props.siteId ) ) || (
+							config.isEnabled( 'jetpack/pijp' ) && 'tryandcustomize' === key && props.hasUnlimitedPremiumThemes
```

Comment:
This kind of logic is exactly what `hideForTheme` is for, so please move it to [`theme-options.js`](https://github.com/Automattic/wp-calypso/blob/c2857ee3ea40f115e1056a945a407b6a53a0309f/client/my-sites/themes/theme-options.js#L106-L117). Otherwise, spreading the hiding logic over several files is going to make it very hard to locate what is filtered where.